src/util/io.c

```diff
@@ -59,6 +59,10 @@ read_all (
                          fd, errno, strerror (errno));
             return recvd_total;
         }
+        if (recvd == 0) {
+            LOG_WARNING ("read on fd %d returned 0 / EOF", fd);
```

Comment:
I would add in the warning output the number of bytes that were attempt to read since that may be relevant information for debugging, something like:

```
LOG_WARNING ("Attempted to read %zd bytes from fd %d, but EOF returned", size, fd);
```

Reply:
done